Packs/CheckpointFirewall/Integrations/CheckPointFirewallV2/CheckPointFirewallV2.py

```diff
@@ -62,12 +62,18 @@ def login(self, username: str, password: str, session_timeout: int, domain_arg:
 
     def restore_sid_from_context_or_login(self, username: str, password: str, session_timeout: int,
                                           domain_arg: str = None):
+
         if sid_from_context := demisto.getIntegrationContext().get('cp_sid'):
-            demisto.debug(f"restore sid: success, setting restored sid on Client (sid={sid_from_context})")
-            self.sid = sid_from_context
-        else:
-            demisto.debug("restore sid: failed to restore, logging in")
-            self.login(username, password, session_timeout, domain_arg)
+            demisto.debug(f"restore sid: found some session id ({sid_from_context}) in context ")
+            if self.test_connection() == 'ok':
+                demisto.debug(f"restore sid: success, setting restored session id ({sid_from_context}) on Client ")
+                self.sid = sid_from_context
+                return
+            else:
+                demisto.debug(f"restore sid: test-connection with session id ({sid_from_context}) failed")
```

Comment:
I think we can mask most of the sid - we only intend to use it for debug purposes, so it's a good practice to not print the full sid.
Please remove the trailing space in line 67.